nil/core-concepts/shards-parallel-execution.mdx

```diff
@@ -0,0 +1,82 @@
+# Shards and parallelized execution
+
+## Definition
+
+Shards are smaller chains that are usually responsible for managing a portion of the global state of the main chain. In =nil;, there exist two types of shards.
+
+* Execution shards process transactions of the accounts allocated to these shards
+* The consensus shard coordinates execution shards and submits ZKPs of the global state to Ethereum
+
+## Execution shards
+
+All work related to processing transactions, storing accounts, and managing the local state occurs at the execution shard level. Execution shards fulfill three other roles listed below.
+
+:::info[Mempools]
+
+Although each shard has its own mempool of transactions, there are no restrictions for the participants on what shard they can cooperate with.
+
+:::
+
+### Message passing checks
+
+To support [**cross-shard communications**](../principles#cross-shard-communication), execution shards must record all necessary outgoing messages from their neighbors.
+
+Every outgoing message has a $destShard$ field, whose value is used in the following function to compute where the message should go.
+
+$$
+nextShard = NextHop(destShard, shardId)
+$$
+
+If $nextShard$ equals the current shard, the message must be included in the block for the block to be considered valid. The block proposer is also responsible for broadcasting a Merkle proof of inclusion to neighboring shards.
+
+### Split/merge conditions checks
+
+Execution shards are [**split and merged**](../principles#execution-shards) depending on how close they are to block capacity.
+
+The protocol defines $N$, the number of replication cycles to pass before a split/merge check, and $T$, the block capacity threshold.
+
+After $N$ replication cycles pass, the following events occur:
+
+* If a shard approaches $T$ blocks, it is split into two
+* If a shard has fewer than $\frac{T}{2}$ blocks, it is merged with another shard
+
+### State consensus checks
+
+In =nil;, validators govern entire neighborhoods of shards based on the Hamming distance between shards.
```

Comment:
I suppose, it should be not "govern" but rather "verify"

Reply:
Fair enough, rephrased